addons/web/static/tests/views/view_dialogs_tests.js

```diff
@@ -243,6 +246,126 @@ QUnit.module('Views', {
         parent.destroy();
     });
 
+    QUnit.test('SelectCreateDialog cascade x2many in create mode', function (assert) {
+        assert.expect(5);
+
+        this.data.product = {
+            fields : {
+                name: {string: "name", type: "char" },
+                partner : {string: 'Doors', type: 'one2many', relation: 'partner'}
+            },
+            records: [
+                {id: 1, name: 'The end'}
+            ]
+        };
+
+        this.data.partner.fields.instrument = {string: 'Instruments', type: 'many2one', relation: 'instrument'};
+
+        this.data.instrument = {
+            fields : {
+                name: {string: "name", type: "char"},
+                badassery: {string: 'level', type: 'many2many', relation: 'badassery', domain: [['level', '=', 'Awsome']]},
+            },
+        };
+
+        this.data.badassery = {
+            fields : {
+                level: {string: 'level', type: "char"},
+            },
+            records: [
+                {id: 1, level: 'Awsome'},
+            ]
+        };
+
+        var form = createView({
+            View: FormView,
+            model: 'product',
+            data: this.data,
+            arch: '<form>' +
+                     '<field name="name"/>' +
+                     '<field name="partner" widget="one2many_list" >' +
+                        '<tree editable="top">' +
+                            '<field name="display_name"/>' +
+                            '<field name="instrument"/>' +
+                        '</tree>' +
+                    '</field>' +
+                  '</form>',
+            res_id: 1,
+            archs: {
+                'partner,false,form': '<form>' +
+                                           '<field name="name"/>' +
+                                           '<field name="instrument" widget="one2many_list" mode="tree"/>' +
+                                        '</form>',
+
+                'instrument,false,form': '<form>'+
+                                            '<field name="name"/>'+
+                                            '<field name="badassery">' +
+                                                '<tree>'+
+                                                    '<field name="level"/>'+
+                                                '</tree>' +
+                                            '</field>' +
+                                        '</form>',
+
+                'badassery,false,list': '<list>'+
```

Comment:
i think that you should use the `<tree>` tag instead of `<list>`.  It is interesting to see that we actually do not validate the name of the tag... maybe we should do that